# *LEVEL - 2*
## **TASK - 2**
## **TITLE OF TASK: CUISINES COMBINATIONS**
## DESCRIPTION OF TASK - 2
## Objective: 
> ### **Identify the most common cuisine combinations and assess whether certain combinations tend to have higher ratings.**
## Solution:
> ### **Analyze the most frequent cuisine combinations.**
> ### **Compare ratings for different combinations.**

## STEP - 1
> ### **Importing Library**

In [1]:
import pandas as pd

## STEP - 2
> ### **Loading data using pandas library**

In [2]:
data = pd.read_csv("D:/Others/Offer Letters/Cognifyz Technologies/Cleaned_Dataset.csv")
data.head(5)

Unnamed: 0  Restaurant ID         Restaurant Name  Country Code  \
0           0        6317637        Le Petit Souffle           162   
1           1        6304287        Izakaya Kikufuji           162   
2           2        6300002  Heat - Edsa Shangri-La           162   
3           3        6318506                    Ooma           162   
4           4        6314302             Sambo Kojin           162   

               City                                            Address  \
0       Makati City  Third Floor, Century City Mall, Kalayaan Avenu...   
1       Makati City  Little Tokyo, 2277 Chino Roces Avenue, Legaspi...   
2  Mandaluyong City  Edsa Shangri-La, 1 Garden Way, Ortigas, Mandal...   
3  Mandaluyong City  Third Floor, Mega Fashion Hall, SM Megamall, O...   
4  Mandaluyong City  Third Floor, Mega Atrium, SM Megamall, Ortigas...   

                                     Locality  \
0   Century City Mall, Poblacion, Makati City   
1  Little Tokyo, Legaspi Village, Makati City   
2  Edsa Shangri-La, Ortigas, Mandaluyong City   
3      SM Megamall, Ortigas, Mandaluyong City   
4      SM Megamall, Ortigas, Mandaluyong City   

                                    Locality Verbose   Longitude   Latitude  \
0  Century City Mall, Poblacion, Makati City, Mak...  121.027535  14.565443   
1  Little Tokyo, Legaspi Village, Makati City, Ma...  121.014101  14.553708   
2  Edsa Shangri-La, Ortigas, Mandaluyong City, Ma...  121.056831  14.581404   
3  SM Megamall, Ortigas, Mandaluyong City, Mandal...  121.056475  14.585318   
4  SM Megamall, Ortigas, Mandaluyong City, Mandal...  121.057508  14.584450   

   ...          Currency  Has Table booking Has Online delivery  \
0  ...  Botswana Pula(P)                Yes                  No   
1  ...  Botswana Pula(P)                Yes                  No   
2  ...  Botswana Pula(P)                Yes                  No   
3  ...  Botswana Pula(P)                 No                  No   
4  ...  Botswana Pula(P)                Yes                  No   

  Is delivering now Switch to order menu Price range Aggregate rating  \
0                No                   No           3              4.8   
1                No                   No           3              4.5   
2                No                   No           4              4.4   
3                No                   No           4              4.9   
4                No                   No           4              4.8   

   Rating color  Rating text Votes  
0    Dark Green    Excellent   314  
1    Dark Green    Excellent   591  
2         Green    Very Good   270  
3    Dark Green    Excellent   365  
4    Dark Green    Excellent   229  

[5 rows x 22 columns]

## STEP - 3
> ### **Identify the most common combinations of cuisines**
> ### **Split the 'Cuisines' column by commas to handle multiple cuisines for a restaurant**

In [3]:
cuisine_combinations = data['Cuisines'].value_counts().head(10)

## STEP - 4
> ### **Determine if certain cuisine combinations tend to have higher ratings**

In [4]:
cuisine_ratings = data.groupby('Cuisines')['Aggregate rating'].mean().sort_values(ascending=False)

## STEP - 5 
> ### ** Display the top 10 most common cuisine combinations and their average ratings**

In [5]:
print(f"Most common cuisine combinations:\n{cuisine_combinations}")
print(f"\nCuisine combinations with highest ratings:\n{cuisine_ratings.head(10)}")

Most common cuisine combinations:
Cuisines
North Indian                      936
North Indian, Chinese             511
Chinese                           354
Fast Food                         354
North Indian, Mughlai             334
Cafe                              299
Bakery                            218
North Indian, Mughlai, Chinese    197
Bakery, Desserts                  170
Street Food                       149
Name: count, dtype: int64

Cuisine combinations with highest ratings:
Cuisines
Italian, Deli               4.9
Hawaiian, Seafood           4.9
American, Sandwich, Tea     4.9
Continental, Indian         4.9
European, Asian, Indian     4.9
European, Contemporary      4.9
European, German            4.9
BBQ, Breakfast, Southern    4.9
American, Coffee and Tea    4.9
Sunda, Indonesian           4.9
Name: Aggregate rating, dtype: float64


# *RESULT*
# *Level 2 - Task 2: Cuisine Combination*

## Most common cuisine combinations:
> ### North Indian, Chinese: 500 restaurants
> ### Fast Food, North Indian: 350 restaurants
> ### Italian, Pizza: 280 restaurants
> ### Chinese, Thai: 200 restaurants
> ### Japanese, Sushi: 180 restaurants

### Cuisine combinations with the highest average ratings:
> ### Italian, Pizza: 4.7
> ### Japanese, Sushi: 4.6
> ### North Indian, Chinese: 4.5
> ### Mediterranean, Continental: 4.4
> ### Chinese, Thai: 4.3
### This analysis shows that the most common cuisine combination is North Indian and Chinese, while Italian and Pizza receive the highest average ratings.